src/effects/rising_bubble.c

```diff
@@ -1,27 +1,60 @@
 #include "common.h"
 #include "effects_internal.h"
 
+void rising_bubble_init(EffectInstance* effect);
+void rising_bubble_update(EffectInstance* effect);
+void rising_bubble_render(EffectInstance* effect);
 void rising_bubble_appendGfx(void* effect);
 
-INCLUDE_ASM(s32, "effects/rising_bubble", rising_bubble_main);
+void rising_bubble_main(s32 arg0, f32 posX, f32 posY, f32 posZ, f32 arg4) {
+    EffectBlueprint effectBp;
+    EffectInstance* effect;
+    RisingBubbleFXData* data;
+    s32 numParts = 1;
 
-void rising_bubble_init(void) {
+    effectBp.unk_00 = 0;
+    effectBp.init = rising_bubble_init;
+    effectBp.update = rising_bubble_update;
+    effectBp.renderWorld = rising_bubble_render;
+    effectBp.unk_14 = 0;
+    effectBp.effectID = EFFECT_RISING_BUBBLE;
+    effect = shim_create_effect_instance(&effectBp);
+    effect->numParts = numParts;
+    data = effect->data.risingBubble = shim_general_heap_malloc(numParts * sizeof(*data));
+    ASSERT(data != NULL);
+
+    data->unk_00 = arg0;
+    data->pos.x = posX;
+    data->pos.y = posY;
+    data->pos.z = posZ;
+    data->unk_14 = 0;
+    data->lifeTime = 0;
+    if (arg4 != 0.0f) {
+        data->unk_10 = ((shim_rand_int(3) * 0.1) + 0.7) * 0.4;
+    } else {
+        data->unk_10 = 1.f;
```

Comment:
I didn't know gcc just flies with this :D
```suggestion
        data->unk_10 = 1.0f;
```

Reply:
lol force of habit